src/app/dataset-view/additional-components/metadata-component/components/add-polling-source/edit-polling-source.service.ts

```diff
@@ -0,0 +1,176 @@
+import { Injectable } from "@angular/core";
+import { zip } from "rxjs";
+import { switchMap, map } from "rxjs/operators";
+import { MetadataBlockFragment } from "src/app/api/kamu.graphql.interface";
+import { BlockService } from "src/app/dataset-block/metadata-block/block.service";
+import { DatasetService } from "src/app/dataset-view/dataset.service";
+import { DatasetHistoryUpdate } from "src/app/dataset-view/dataset.subscriptions.interface";
+import { AppDatasetSubscriptionsService } from "src/app/dataset-view/dataset.subscriptions.service";
+import { DatasetInfo } from "src/app/interface/navigation.interface";
+import { parse } from "yaml";
+import {
+    EditFormParseType,
+    EditFormType,
+    FetchKind,
+    MergeKind,
+    NameValue,
+} from "./add-polling-source-form.types";
+import { SetPollingSourceSection } from "src/app/shared/shared.types";
+import { FormArray, FormBuilder, FormGroup } from "@angular/forms";
+import { RxwebValidators } from "@rxweb/reactive-form-validators";
+
+@Injectable({
+    providedIn: "root",
+})
+export class EditPollingSourceService {
+    private currentPage = 0;
+    private historyLength = 200;
+    public history: DatasetHistoryUpdate;
+
+    constructor(
+        private appDatasetService: DatasetService,
+        private blockService: BlockService,
+        private appDatasetSubsService: AppDatasetSubscriptionsService,
+        private fb: FormBuilder,
+    ) {}
+
+    public parseEventFromYaml(event: string): EditFormType {
+        const editFormParseValue = parse(event) as EditFormParseType;
+        return editFormParseValue.content.event;
+    }
+
+    public patchFormValues(
+        sectionForm: FormGroup,
+        editFormValue: EditFormType,
+        groupName: SetPollingSourceSection,
+    ): void {
+        switch (groupName) {
+            case SetPollingSourceSection.FETCH: {
+                this.patchFetchStep(sectionForm, editFormValue);
+                break;
+            }
+            case SetPollingSourceSection.READ: {
+                this.patchReadStep(sectionForm, editFormValue);
+                break;
+            }
+            case SetPollingSourceSection.MERGE: {
+                this.patchMergeStep(sectionForm, editFormValue);
+                break;
+            }
+        }
+    }
+
+    public getSetPollingSourceAsYaml(info: DatasetInfo) {
+        return this.appDatasetService
+            .requestDatasetHistory(info, this.historyLength, this.currentPage)
+            .pipe(
+                switchMap(
+                    () => this.appDatasetSubsService.onDatasetHistoryChanges,
+                ),
+                map((h: DatasetHistoryUpdate) => {
+                    this.history = h;
+                    const lastBlock = h.history.filter(
+                        (item: MetadataBlockFragment) =>
+                            item.event.__typename === "SetPollingSource",
+                    )[0];
```

Comment:
This should go in loop reading history page by page until:
 - SetPollingSource event is found (it may not be found)
 - we reached the end of history